cli/src/ui.rs

```diff
@@ -580,16 +578,15 @@ impl Ui {
         let mut buf = String::new();
         io::stdin().read_line(&mut buf)?;
 
-        if let Some(trimmed) = buf.strip_suffix('\n') {
-            buf = trimmed.to_owned();
-        } else if buf.is_empty() {
-            return Err(io::Error::new(
+        let buf = buf.strip_suffix('\n').map(|s| s.to_owned()).unwrap_or(buf);
```

Comment:
nit: `buf.truncate(trimmed.len())` could also be used.

btw, if `buf == "\n"`, an empty string should be returned. It's not an EOF.